google/cloud/storage/internal/object_requests.cc

```diff
@@ -122,7 +122,12 @@ internal::nl::json ObjectMetadataJsonForUpdate(ObjectMetadata const& meta) {
   SetIfNotEmpty(metadata_as_json, "contentLanguage", meta.content_language());
   SetIfNotEmpty(metadata_as_json, "contentType", meta.content_type());
 
-  metadata_as_json["eventBasedHold"] = meta.event_based_hold();
+  if (meta.event_based_hold()) {
+    metadata_as_json["eventBasedHold"] = true;
```

Comment:
Might it be worth growing a `SetIfTrue()` to match `SetIfNotEmpty()`?

Reply:
I looked around and could not find any other usage.  I am going to leave as-is for now.